test/performance/odp_ipsec.c

```diff
@@ -665,13 +678,8 @@ run_measure_one(ipsec_args_t *cargs,
 				odp_packet_free(pkt);
 				break;
 			}
-			if (odp_packet_has_error(out_pkt)) {
-				odp_ipsec_packet_result_t result;
+			check_ipsec_result(out_pkt);
```

Comment:
I had noticed this discrepancy when I added the burst support. The way I understood, odp_ipsec_result() call in ASYNC path is mandatory to have post-processing done. For SYNC, it's not mandatory unless to verify that IPsec operation has been completed successfully.

The thing I wasn't sure about is, do we expect performance application also to call odp_ipsec_result() and check the results? If yes, why not make odp_ipsec_result() mandatory for both SYNC & ASYNC?

Usage of odp_packet_has_error() is wrong anyway. So I agree with the overall fix.

Reply:
Since odp_ipsec_result() is practically always used, it makes sense for this test app to use it too even when not mandatory. Even in the async case the call is not mandatory overall but just needs to be done before packet metadata or packet data is accessed. I am not sure adding that requirement to the sync case would have much benefit, but it would add an additional restriction.

Reply:
> I am not sure adding that requirement to the sync case would have much benefit, but it would add an additional restriction.
I don't think it's a restriction if we agree that the API is used "practically always". Currently, with SYNC APIs, implementation would require executing post-processing from the odp_ipsec_in/out APIs and when odp_ipsec_result() is called, it would need additional logic to skip post-processing.

My suggestion is to avoid extra checks in the data path.

Reply:
Ok, it could save one check there. But it would require an API change which would have to be handled separately. It is not really related to this PR.

Reply:
Agreed.